Enable rainbow curve

zhx828 opened the pull request:
Heroku instance: https://rainbow-survival.herokuapp.com/

Issues:
- [x] Disable survival icon if there is no survival data available

zhx828 committed:
Introduce `groups` concept into survival

Calculate curve data in survival template instead of survival model

zhx828 committed:
Enable rainbow survival for pie chart

zhx828 committed:
Enable rainbow survival for bar chart

zhx828 committed:
Add tooltip for survival plot

zhx828 committed:
Change the all cases tooltip name

zhx828 committed:
Enable rainbow survival for tables

zhx828 committed:
Enable highlight in survival tooltip

zhx828 committed:
Fix issue #385

zhx828 committed:
Fix issue #379

yichaoS commented on app/scripts/views/components/dataTable/tableViewTemplate.js:
```diff
@@ -169,6 +198,17 @@
+        }
+
+        var categories = this.chartInst.getCurrentCategories();
+        if (_.isArray(categories) && (categories.length > 1 && categories.length <= 20)) {
```
Do you think maybe we can put all the settings, like "20" here into one place?

zhx828 replied:
Good point. I can do that

yichaoS commented on app/scripts/views/components/pieChart/pieChartTemplate.js:
```diff
@@ -146,6 +174,11 @@
       });
+
+      // Disable rainbow survival if only one group present
+      if (_self.piechart.getCurrentCategories().length < 2 || _self.piechart.getCurrentCategories().length > 20) {
```
Same here, this "20" as settings, maybe can all go to one place?

yichaoS commented on app/scripts/model/dataProxy.js:
```diff
@@ -1282,47 +1282,48 @@ window.DataManagerForIviz = (function($, _) {
-              _map[_panelMeta.stableId]["sel_samples"] = (_panelMeta.samples);
+              var _sorted = (_panelMeta.samples).sort();
+              _map[_panelMeta.stableId].samples = (_sorted);
+              _map[_panelMeta.stableId].sel_samples = (_sorted);
```
Are we keeping this ()? I know I probably wrote that (my fault).

kalletlak replied:
better to cleanup

zhx828 committed:
Remove unnecessary parenthesis

zhx828 committed:
Change vars.json to configs.son; Customize number of survival curves

zhx828 committed:
Show survival icon only if study has survival data

zhx828 committed:
Add `No data available`

This fixes https://github.com/cBioPortal/iViz/issues/422
Related to https://github.com/cBioPortal/cbioportal/issues/2247

zhx828 committed:
Fix pie slice highlight bug

zhx828 committed:
Solve the icon shifting issue

Related to https://github.com/cBioPortal/iViz/issues/189

zhx828 committed:
Fix issues in the comments

kalletlak commented on app/scripts/views/components/dataTable/tableViewTemplate.js:
```diff
@@ -12,6 +12,7 @@
     ':display-name="displayName" :chart-ctrl="chartInst"' +
     ':has-chart-title="true" :groupid="attributes.group_id" ' +
     ':reset-btn-id="resetBtnId" :chart-id="chartId" :attributes="attributes" ' +
+    ':show-survival-icon.sync="showSurvivalIcon"' +
```
show we have button description of this some thing like `alt ="rainbow survival"`

zhx828 replied:
I don't get your point

kalletlak replied:
basically a title for those icons

kalletlak commented on app/scripts/views/components/barChart/barChartTemplate.js:
```diff
@@ -43,6 +44,7 @@
           transitionDuration: iViz.opts.dc.transitionDuration
         },
         opts: {},
+        numOfSurvivalCurveLimit: iViz.opts.numOfSurvivalCurveLimit || 20,
```
since we already set iViz.opts.numOfSurvivalCurveLimit, why to check it again?

zhx828 replied:
This is customizable in configs.json

kalletlak commented on app/scripts/views/components/dataTable/tableViewTemplate.js:
```diff
@@ -39,7 +40,9 @@
-        genePanelMap: {}
+        showSurvivalIcon: false,
+        genePanelMap: {},
+        numOfSurvivalCurveLimit: iViz.opts.numOfSurvivalCurveLimit || 20
```
even here

kalletlak commented on app/scripts/views/components/pieChart/pieChartTemplate.js:
```diff
@@ -42,7 +43,9 @@
         filtersUpdated: false,
-        addingChart: false
+        addingChart: false,
+        numOfSurvivalCurveLimit: iViz.opts.numOfSurvivalCurveLimit || 20,
```
here too

kalletlak commented on app/styles/partials/_jsonVars.scss:
```diff
@@ -1,4 +1,6 @@
-$grid-h-2:350px;
\ No newline at end of file
+$grid-h-2:350px;
+$numOfSurvivalCurveLimit:20;
```
where is this field used?

zhx828 replied:
This is automatically generated based on configs.json As long as you define a variable in there, there will be a variable in _jsonVars.scss

zhx828 committed:
Add alt for image
